src/games/strategy/engine/data/Unit.java

```diff
@@ -95,6 +95,13 @@ public boolean equals(final Object o) {
     return this.m_uid.equals(other.m_uid);
   }
 
+  public boolean hasSamePropertiesAndState(final Unit unit) {
```

Comment:
Sounds reasonable. the method name does not jive super well with me: `hasSamePropertiesAndState`
same properties is often the same as matching state. Plus we are matching type, owner, and number of hits. I somewhat assume there are more properties, and more state to a Unit object than that. 

A comment you had while using this was that we were looking for equivalent units. Perhaps name this method `isEquivalent`? I think then this method makes a lot of sense, and is a reasonable definition for equivalence (same owner, type, and damage). Is movement another component that should be included?